qiskit/result/mitigation/correlated_readout_mitigator.py

```diff
@@ -122,7 +124,7 @@ def expectation_value(
             diagonal = str2diag(diagonal)
 
         # Apply transpose of mitigation matrix
-        coeffs = mit_mat.T.dot(diagonal)
+        coeffs = mit_mat.T.dot(diagonal)  # TODO: won't work if diagonal is Callable or dict?
```

Comment:
The case of diagonal being `Callable` or `dict` is not explained anywhere in documentation (even though it is mentioned there) and is not tested. Not sure what it means.